client/components/common/SelectElement/index.vue

```diff
@@ -2,9 +2,26 @@
   <tailor-dialog
     @click:outside="close"
     :value="true"
-    header-icon="mdi-transit-connection-variant"
+    :header-icon="headerIcon"
+    scrollable
     width="650">
-    <template v-slot:header>{{ heading }}</template>
+    <template v-slot:header>
+      {{ heading }}
+      <v-btn
+        v-if="selectedActivity && !loadingContent"
+        @click="toggleSelectAll"
+        text dark
+        class="float-right">
+        <template v-if="allElementsSelected">
+          <v-icon class="mr-2">mdi-checkbox-multiple-blank-outline</v-icon>
```

Comment:
Instead of repeating the same block consider creating a computed returning icon and label based on state.